### DISCUSSION TITLE
GTA3: Continuous crashes after exiting the game via alt+f4?

### DISCUSSION THREAD

**Discussion by eMOElein** [cleolibrary/CLEO-Redux#75] [Q&A] [answered]
I am currently building a JS framework that adds an additional layer of abstraction over CLEO Redux making it easier for me to design missions and new content for the game. Currently testing it with OG GTA3.

I experience this weird behaviour where I exit the game with alt+f4 and when starting the game again I get constant crashes when my code tries to add a blip for coordinates. cleo_redux.log says:
18:45:32 [INFO] LastCommandDebugInfo { name: "ADD_BLIP_FOR_COORD", input: [Float(-644.05963), Float(-42.435585), Float(18.061378)], output: [], if_result: false }

This also happens for character blips.
Spawning cars, characters, wapon pickups etc. does not cause any issues at all.

The strange thing is that the issue even survives multiple restarts of the game.
I can fix it by loading into the game again and hammering the escape button to open and close the menu multiple times.
Now the issue is constantly gone and won't reappear until I exit with alt+f4 again. Don't know why that works.

Are there any known issues when the game is not closed properly? Maybe some caching or finalizers that can't work correctly when closing the game in some other way than through the menu?
I know taht the issue can be easily circumvented by just not closing the game wih alt+f4 but I am curious to learn if there are some technical reasons that might explain this behaviour.

I added a dump from the crashinfo plugin but I could not find anything for the specific address where the error occurs.
If someone wants to take a look at my code I can give you access to my github repo however everything is still very work in progress atm.

[dump.txt](https://github.com/cleolibrary/CLEO-Redux/files/9853621/dump.txt)

**Comment by x87** [comment]
Thanks for detailed description! I haven't heard of nor experienced that particular issue before, so I need to try to reproduce it on my end.

**Reply by eMOElein** [reply]
Sorry, Layer 8 problem.
I just realized that my installation was not properly setup.
Usually using JSGME to keep everything as vanilla as possible when developing but i had this old gang hostility cs script in my script folder that I placed there manually. No more issues after removing it.
CLEO Redux is working asbolutely fine!

**Reply by x87** [reply]
Good to know :) don't forget to share your framework with me when it is ready!

**Reply by eMOElein** [reply]
Still working on the framework.
Unfortunately I still didn't get completely rid of my crashes but I can reliable reproduce them on my end now.
I wrote a little sample script. I testet on OG GTA3. Vanilla as possible.
It also crashes my Vice City installation.
My San Andreas installation which is heavily modded has no issues at all.
I used my original german disc version to install the games.

The script constantly draws a blip and removes it again with the next cycle every 50ms.
Just hit the T key and start walking. Sometimes it took me up to a minute before my crash happened sometimes it crashed after a few seconds.
Vice City tends to crash earlier than GTA3.
Are you able to reproduce this on your end?

[crashReproducer.txt](https://github.com/cleolibrary/CLEO-Redux/files/10035254/crashReproducer.txt)

**Comment by x87** [comment, marked as answer]
@eMOElein thanks for providing a script. I was able to identify the problem. Can you please test this build and see if you still get crashes?

[cleo_redux.zip](https://github.com/cleolibrary/CLEO-Redux/files/10045610/cleo_redux.zip)

**Reply by eMOElein** [reply]
Awesome!
I tested a few times with the sample script for several minutes and tried some of my debug missions in GTA3.
I did a quick test with Vice City as well.
Works like a charm. Many thanks.

I'll do some more extensive testing in the next days with both games.

**Reply by x87** [reply]
Great news!
